bolt-sidecar/src/primitives/commitment.rs

```diff
@@ -173,10 +176,29 @@ impl InclusionRequest {
     /// Validates the priority fee against a minimum priority fee.
     /// Returns `true` if the "effective priority fee" is greater than or equal to the set minimum
     /// priority fee, `false` otherwise.
-    pub fn validate_min_priority_fee(&self, max_base_fee: u128, min_priority_fee: u128) -> bool {
-        self.txs.iter().all(|tx| {
-            tx.effective_tip_per_gas(max_base_fee).map_or(false, |tip| tip >= min_priority_fee)
-        })
+    /// Returns an error if min priority fee cannot be calculated.
+    pub fn validate_min_priority_fee(
+        &self,
+        pricing: &PreconfPricing,
+        preconfirmed_gas: u64,
+        max_base_fee: u128,
+    ) -> Result<bool, PricingError> {
+        // Each included tx will move the price up
+        // So we need to calculate the minimum priority fee for each tx
+        let mut local_preconfirmed_gas = preconfirmed_gas;
+        for tx in &self.txs {
+            // Calculate minimum required priority fee for this transaction
+            let min_priority_fee =
+                pricing.calculate_min_priority_fee(tx.gas_limit(), preconfirmed_gas)?;
```

Comment:
Can this function return  an `u128`?

Reply:
Yes it could, but why is it necessary?
For 30M gas the expected result is [613_499_092..19_175_357_339]

Reply:
Because on the line below you do a conversion to an u128, but it's not a strong requirement tho